onnx_tf/frontend.py

```diff
@@ -189,40 +188,54 @@ def tensorflow_graph_to_onnx_graph(cls,
                 data_type=node.attr["dtype"],
                 dims=shape,
                 vals=raw_values))
-        input_proto = make_tensor_value_info(name, node.attr["dtype"],
-                                             shape)
+        input_proto = make_tensor_value_info(name, node.attr["dtype"], shape)
         inputs_proto.append(input_proto)
       else:
-        handler_name = "handle_" + op_name_to_lower(node.op)
-
-        versions = frontend_tf_opset_version[op_name_to_lower(node.op)]
-
-        assert isinstance(
-            opset, int
-        ) and (opset <= defs.onnx_opset_version()) and (
-            opset >= 0
-        ), "Opset should be an int less than or equal to {}, but {}: {}".format(
-            defs.onnx_opset_version(),
-            type(opset).__name__, opset)
-
-        if opset == 0:
+        splitted_op_name = node.op.split(".")
+        op_domain = "" if len(splitted_op_name) == 1 else ".".join(
+            splitted_op_name[:-1])
+        op_name = splitted_op_name[-1]
+
+        handler_name = "handle_" + op_name_to_lower(op_name)
+
+        # TODO per domain frontend_tf_opset_version?
+        versions = frontend_tf_opset_version[op_name_to_lower(op_name)]
+
+        opset_dict = {}
+        onnx_domain = defs.ONNX_DOMAIN
+        for domain, version in opset:
+          if domain == "ai.onnx":
+            domain = ""
+          opset_dict[domain] = version
+          defs.ONNX_DOMAIN = domain
+          assert isinstance(
+              version, int
+          ) and (version <= defs.onnx_opset_version()) and (
+              version >= 0
+          ), "Opset should be an int less than or equal to {}, but {}: {}".format(
+              defs.onnx_opset_version(), type(version), version)
+          defs.ONNX_DOMAIN = onnx_domain
+
+        opset_ver = opset_dict[op_domain]
+        if opset_ver == 0:
           version = max(versions)
         else:
-          versions = sorted(versions + [opset])
+          versions = sorted(versions + [opset_ver])
           version = versions[
-              max([i for i, v in enumerate(versions) if v == opset]) - 1]
+              max([i for i, v in enumerate(versions) if v == opset_ver]) - 1]
 
+        camel_domain = "".join(w.title() for w in op_domain.split("."))
         frontend_ver = 'frontend_v{}'.format(version)
         frontend = cls.frontend_version_cache.setdefault(
-            frontend_ver,
-            importlib.import_module('onnx_tf.frontends.' + frontend_ver)
-            .TensorflowFrontend)
+            frontend_ver + "_" + op_domain,
```

Comment:
Can we add an err msg when the domain is unknown to us?

Reply:
Sure. I've added NotImplementedError.